websocket_service_test.go

```diff
@@ -455,6 +456,37 @@ func (s *websocketServiceTestSuite) TestWsUserDataServe() {
     }`))
 }
 
+func (s *websocketServiceTestSuite) testWsFutureUserDataServe(data []byte) {
+	fakeErrMsg := "fake error"
+	s.mockWsServe(data, errors.New(fakeErrMsg))
+	defer s.assertWsServe()
+
+	doneC, stopC, err := WsFutureUserDataServe("listenKey", func(event []byte) {
+		s.r().Equal(data, event)
```

Comment:
can we use s.assertWsFutureUserDataEqual to check as L546?

Reply:
@adshao I'm not sure what you'd like for changes. I was actually copy / pasting the original `WsUserDataServe` (lines 418 to 472). Shall it not be based on the same?

Reply:
It's ok.